src/ethereum_test_tools/spec/eof/eof_test.py

```diff
@@ -48,6 +84,51 @@ def make_eof_test_fixture(
                 }
             }
         )
+        eof_parse = EOFParse()
+        for _, vector in fixture.vectors.items():
+            expectedResult = vector.results.get(str(fork))
+            result = eof_parse.run(input=str(vector.code))
+            self.verify_result(result, expectedResult, vector.code)
+
+        return fixture
+
+    def verify_result(self, result: CompletedProcess, expectedResult: Result, code: Bytes):
+        parser = EvmoneExceptionParser()
+        res_error = result.stdout.replace("\n", "")
+        if expectedResult.exception is None:
+            if "OK" not in result.stdout:
+                msg = "Expected eof code to be valid, but got an exception:"
+                formatted_message = (
+                    f"{msg} \n"
+                    f"{code} \n"
+                    f"Expected: No Exception \n"
+                    f"     Got: {parser.rev_parse_exception(res_error)} ({res_error})"
+                )
+                raise Exception(formatted_message)
+        else:
+            if "OK" in res_error:
+                expRes = expectedResult.exception
+                msg = "Expected eof code to be invalid, but got no exception from eof tool:"
+                formatted_message = (
+                    f"{msg} \n"
+                    f"{code} \n"
+                    f"Expected: {expRes} ({parser.parse_exception(expRes)}) \n"
+                    f"     Got: No Exception"
+                )
+                raise Exception(formatted_message)
+            else:
+                expRes = expectedResult.exception
+                if expRes == parser.rev_parse_exception(res_error):
+                    return
+
+                msg = "EOF code expected to fail with a different exception, than reported:"
+                formatted_message = (
+                    f"{msg} \n"
+                    f"{code} \n"
+                    f"Expected: {expRes} ({parser.parse_exception(expRes)}) \n"
+                    f"     Got: {parser.rev_parse_exception(res_error)} ({res_error})"
+                )
+                raise Exception(formatted_message)
```

Comment:
this exception verification feature can be done in the same manner for state tests/bc tests with different tools. 
would be nice to see it in consume.  and also when generating the tests via t8n, the exception string from t8n can be verified against our exception maps

Reply:
Yup, definitely!